src/test/java/com/github/checkstyle/regression/git/DiffParserTest.java

```diff
@@ -0,0 +1,60 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.checkstyle.regression.git;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+
+import org.eclipse.jgit.lib.Repository;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.github.checkstyle.regression.internal.GitUtils;
+
+public class DiffParserTest {
+    private Repository repository;
+
+    @Before
+    public void setUp() throws Exception {
+        repository = GitUtils.newRepository();
+        final File helloWorld = GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+        GitUtils.createNewBranchAndCheckout(repository, "foo");
+        Files.write(helloWorld.toPath(), "hello world!".getBytes(), StandardOpenOption.APPEND);
+        GitUtils.addAllAndCommit(repository, "append text to HelloWorld");
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        repository.close();
+    }
+
+    @Test
+    public void testParse() throws Exception {
```

Comment:
We should include all types of differences in git.
New file, removed file, renamed file, moved file, chmod change (linux only).